File: azkar/src/Core/Tick.swift
/// What a reminder tick does: show a card, or skip it and say why.
enum TickDecision: Equatable {
  case show
  case skip(String)
}

func decideTick(paused: Bool, locked: Bool, quiet: Bool, stackCount: Int, maxStack: Int) -> TickDecision {
  if paused { return .skip("paused") }
  if locked { return .skip("screen locked") }
  if quiet { return .skip("quiet hours") }
  if stackCount >= maxStack { return .skip("stack full") }
  return .show
}

File: azkar/kotlin/shared/src/commonMain/kotlin/com/wildduck/azkar/core/Tick.kt
package com.wildduck.azkar.core

/** What a reminder tick does: show a card, or skip it and say why. */
sealed interface TickDecision {
    data object Show : TickDecision

    data class Skip(val reason: String) : TickDecision
}

fun decideTick(paused: Boolean, locked: Boolean, quiet: Boolean, stackCount: Int, maxStack: Int): TickDecision =
    when {
        paused -> TickDecision.Skip("paused")
        locked -> TickDecision.Skip("screen locked")
        quiet -> TickDecision.Skip("quiet hours")
        stackCount >= maxStack -> TickDecision.Skip("stack full")
        else -> TickDecision.Show
    }
